modules/compliance-tests/src/smithy4s/compliancetests/internals/CanonicalSmithyDecoder.scala

```diff
@@ -40,7 +39,7 @@ object CanonicalSmithyDecoder {
       schema: Schema[A]
   ): Document.Decoder[A] = {
     decoder.fromSchema(
-      schema.transformHintsTransitively(_.filter(_.keyId == IntEnum.id))
```

Comment:
we were not sure about this change, it didn't break any test in this module but I think there's only one test of regular string enumerations, we could add a similar one for int enumerations but not sure I get the point of these tests